benchmarks/README.md

```diff
@@ -0,0 +1,83 @@
+# Description
+
+If you want to execute this module, you can do so by using the following command:
+
+```bash
+python -m benchmarks
+```
+
+## Benchmark
+
+A combination of all benchmark related data. A callable is used to perform and hide the
+benchmarked code. The name serves as the unique identifier of the benchmark. Note that
+this identifier is also used to store the results. Therefore, any change will be
+considered a new benchmark when storing results. The description is currently, the
+description is automatically generated based on the `__doc__` attribute of
+the `function` callable. A full code example can be found in the
+`domains/synthetic_2C1D_1C.py` file.
+
+## BenchmarkSettings
+
+The settings object is used to parameterize the benchmark. It is an abstract base
+class that can be extended by the user to provide additional information for the
+benchmark. The only required attribute is `random_seed`, which is used to seed the
+entire call of the benchmark function. Currently, the following settings are available:
+
+### ConvergenceExperimentSettings
+
+The settings object is used to parameterize the convergence experiment benchmarks
+and holds information used for BayBE scenario executions. Please refer to the BayBE
+documentation for more information about experiment simulations.
+
+## Result
+
+The result object encapsulates all execution-relevant information of the benchmark
+and represents the result of the benchmark function, along with state information
+at the time of execution.
+
+### ResultMetadata
+
+The metadata is the wrapper to hold the described information about the benchmark
+at runtime. A combination of the benchmark identifier and the metadata is meant to
+describe the conducted result uniquely under the assumption that equal benchmarked code
+states are also equally representative due to the fixed random seed.
+
+### Add your benchmark to the benchmarking module
+
+In the last step, your benchmark object has to be added to the benchmarking module.
+This is done by adding the object to the `BENCHMARKS` list in the `__init__.py` file
+in the `domains` folder. The `BENCHMARKS` list is a list of all benchmark objects that
+should be executed when running the benchmarking module.
+
+## Persisting Results
+
+Results are stored automatically. Since multiple storage types are provided with
+different requirements and compatibilities, the `PathConstructor` class is used to
+construct the identifier for the file. For example `S3ObjectStorage` is used to
+store the results in an S3 bucket which separates the key by `/` but does not create
+real folders while the usual local persistence creates a file with a `_` so that folder
+creation is not necessary. The class handling the storage of the resulting object get
+this `PathConstructor` and use it in the way it needs the identifier to be.
+There are currently two storage types available:
```

Comment:
"The following types of storage are available:" - This is just a more robust reformulation as it works for an arbitrary number of storage types